index.js

```diff
@@ -1,184 +1,249 @@
-// updateGrade(student1, 92)
+ 
+ 
+// console.log("%c----------", "color: red")
+ 
+function updateGrade(student, updatedGrade){
+ student.grade = updatedGrade
+}
+ 
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+updateGrade(student1, 67)
 // console.log(student1)
 // // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
-
-
-
+ 
+ 
+ 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
 // console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
+ 
+const users = [
+ {
+   name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
+ },
+ {
+   name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+ }
+]
+ 
 // // => "Duane"
 // => "Cell: 555-123-4567"
 // => "Office: 555-456-7890"
 // => "Liza"
 // => "Cell: 555-234-5678"
 // => "Office: 555-567-1234"
-// console.log("%c----------", "color: red") 
-
-
+// console.log("%c----------", "color: red")
+ 
+function printNameAndPhones(users){
+for (let i = 0 < users.length; i++) {
+  console.log(user.name)
+  console.log(user.phones)
+}
+}
+// Uncaught SyntaxError: Unexpected token ')' -- can't figure this out! UGH.
+ 
+ 
+console.log(printNameAndPhones(users))
+ 
+// **RUBY CODE **
+// def print_name_and_phones(users)
+//   users.each do |user_hash|
+//     puts user_hash[:name]
+//     puts "Cell: #{user_hash[:phones][:cell]}"
+//     puts "Office: #{user_hash[:phones][:office]}"
+//   end
+// end
+ 
+// users = [
+//   {
+//     name: "Duane",
+//     phones: {
+//       cell: "555-123-4567",
+//       office: "555-456-7890",
+//     }
+//   },
+//   {
+//     name: "Liza",
+//     phones: {
+//       cell: "555-234-5678",
+//       office: "555-567-1234"
+//     }
+//   }
+// ]
+ 
+ 
 // ***** Callbacks *****
-
+ 
 function myMap(array, callback) {
-  const result = []
-  for (let i = 0; i < array.length; i++) {
-    const element = array[i]
-    const value = callback(element)
-    result.push(value)
-  }
-  return result
+ const result = []
+ for (let i = 0; i < array.length; i++) {
+   const element = array[i]
+   const value = callback(element)
+   result.push(value)
+ }
+ return result
 }
-
+ 
 // ***** Callbacks - Question 1 *****
-
+ 
 // *** Uncomment the lines below to test
 // console.log("%cCallbacks - Question 1", "color: red")
-
-// console.log(myMap([1,2,3,4,5], triple)) 
+ 
+// console.log(myMap([1,2,3,4,5], triple))
 // // => [3,6,9,12,15]
-
-// console.log(myMap([2,4,6,8], triple))   
+ 
+// console.log(myMap([2,4,6,8], triple))  
 // // => [6,12,18,24]
-// console.log("%c----------", "color: red") 
-
-
+// console.log("%c----------", "color: red")
+ 
+function triple(number){
+ number * 3
+}
+ 
+ 
```

Comment:
Your code is super close. Just remember that in JS we need to be explicit with our returns. Currently, your `triple` function is returning `undefined` because we have not told it what to return.

try adding a return to line 174
function triple(number){
 return number * 3
}